homeassistant/components/awattar/sensor.py

```diff
@@ -0,0 +1,225 @@
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_6_hrs_start",
+        translation_key="slot_6_hrs_start",
+        service_type="energy",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.awattar.best_slot(6).start_datetime,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_2_hrs_price",
+        translation_key="slot_2_hrs_price",
+        service_type="energy",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.best_slot(2).price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_3_hrs_price",
+        translation_key="slot_3_hrs_price",
+        service_type="energy",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.best_slot(3).price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_4_hrs_price",
+        translation_key="slot_4_hrs_price",
+        service_type="energy",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.best_slot(4).price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_5_hrs_price",
+        translation_key="slot_5_hrs_price",
+        service_type="energy",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.best_slot(5).price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_6_hrs_price",
+        translation_key="slot_6_hrs_price",
+        service_type="energy",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.best_slot(6).price_per_kWh,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up aWATTar Sensors based on a config entry."""
+    coordinator: AwattarDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        AwattarSensorEntity(
+            coordinator=coordinator,
+            description=description,
+        )
+        for description in SENSORS
+    )
+
+
+class AwattarSensorEntity(
+    CoordinatorEntity[AwattarDataUpdateCoordinator], SensorEntity
+):
+    """Defines a aWATTar sensor."""
+
+    _attr_has_entity_name = True
+    _attr_attribution = "Data provided by aWATTar"
```

Comment:
Is a attribution needed?

Reply:
First of all, thanks for all your suggestions, i am pretty overwhelmed on how this whole project runs ! - ok, its my first approach on submitting something to such a ahuge project !

what does attribution generally mean ? do i need to set it when the API provider requires me to ? i basically took the easyenergy integration as base *sigh*

Reply:
It's no problem 😊 we are there to help! And no need to be overwhelmed, you can address the points step by step if you want to.

Normally the API provider has some terms of use. And some have a clause that in order to use the API you need to credit them, in some cases even writing what you should use as an attribution. But when there is no such clause, it is not needed.

Reply:
Exactly what @autinerd said. The problem with the attribution is that it's not translatable, so it would be in English even though the rest of the UI is in your chosen language.

Reply:
Thanks for your answers, on the page they say that one can use the API for free with fair-use (max. 100 queries a day). Thats why i only query every hour. So i think i can remove the attribution.

But, is there is a way to query the api once on integration start and then every day at 14:00 + some random delay as the data only changes at 14:00 ? and how would this be implemented ?

Another question regarding entity-id's: mine look this: 

`sensor.energy_market_price_current_hour`

whereas easyenergy's integration entity is: 

`sensor.easyenergy_today_energy_usage_current_hour`

Shouldn't they be unique (to have all energy providers providing the same entity-ids, but then you cannot have more than one energy provider) ? 

Or should i add `awattar_` to the entity-id to distinguish them so that every integration providing energy data has different - and unique entity-id's ?

I think the other points (except testing) i already solved thanks to your suggestions :)